app/services/puppet_fact_parser.rb

```diff
@@ -11,8 +11,8 @@ def operatingsystem
       args = {:name => os_name, :major => major, :minor => minor}
       os = Operatingsystem.find_or_initialize_by(args)
       if os_name[/debian|ubuntu/i] || os.family == 'Debian'
-        if facts.dig(:os, :distro, :codename).presence || facts[:lsbdistcodename]
-          os.release_name = facts.dig(:os, :distro, :codename).presence || facts[:lsbdistcodename]
+        if distro_codename
+          os.release_name = distro_codename
         elsif os.release_name.blank?
           os.release_name = 'unknown'
         end
```

Comment:
Not really because it should only be set to `unknown` if it was blank. The distro code name is an optional fact that typically depends on `lsb-release` to be installed (maybe no longer in Facter 4, not sure). You don't want a host where that's not installed to reset a correct value back to `unknown`.

In the future I think the `unknown` part should be dropped, but that's out of scope for this PR.

Reply:
> In the future I think the `unknown` part should be dropped, but that's out of scope for this PR.

For what it's worth, I dropped it in https://github.com/theforeman/foreman/pull/8450, specifically  f667ef7e85e0d5ba5b1be0ee53840d31897379f1. I'd be happy to submit that as a standalone PR with its own issue.